### PR TITLE
Created pull request template

### PR DESCRIPTION
**Pull request by francescalb**
Created a pull request template in the .github folder.

This is intended to be used by the person creating the pull request prior to asking for review and as a help for the reviewer.

It addresses issue 190, which asks for a checklist for the reviewer. This pull request makes #191 redundant.

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# Description:
+_Summary of change, including the issue(s) to be adressed._
+
+## Type of change:
+_Put an `x` in the box that applies._
```
Comment (CasperWA) [inline .github/pull_request_template.md:5]:
```suggestion
<!-- Put an `x` in the box that applies. -->
```

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
+# Description:
+_Summary of change, including the issue(s) to be adressed._
```
Comment (CasperWA) [inline .github/pull_request_template.md:2]:
```suggestion
<!-- Summary of change, including the issue(s) to be addressed. -->
```

File: .github/pull_request_template.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+## Type of change:
+_Put an `x` in the box that applies._
+- [ ] Bug fix.
+- [ ] New feature.
+- [ ] Documentation update.
+
+## Checklist:
+_Put an `x` in the boxes that apply. You can also fill these out after creating the PR._
```
Comment (CasperWA) [inline .github/pull_request_template.md:11]:
```suggestion
<!-- Put an `x` in the boxes that apply. You can also fill these out after creating the PR. -->
```

File: .github/pull_request_template.md [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+- [ ] Are comments for humans to read, not computers to disregard?
+- [ ] Does a new feature has an accompanying new test in the CI/CD?
+- [ ] Has the documentation been updated as necessary?
+- [ ] Does this close the issue?
+- [ ] Is the change limited to the issue?
+- [ ] Are errors handled for all outcomes?
+- [ ] Does the new feature provide new restrictions on dependecies, and if so is this documented?
+
+## Comments:
+_Additional comments here, including clarifications on checklist if applicable._
```
Comment (CasperWA) [inline .github/pull_request_template.md:25]:
```suggestion
<!-- Additional comments here, including clarifications on checklist if applicable. -->
```

File: .github/pull_request_template.md [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+## Type of change:
+_Put an `x` in the box that applies._
+- [ ] Bug fix.
+- [ ] New feature.
+- [ ] Documentation update.
+
+## Checklist:
+_Put an `x` in the boxes that apply. You can also fill these out after creating the PR._
+
+This checklist can be used as a help for the reviewer.
```
Comment (CasperWA) [inline .github/pull_request_template.md:13]:
```suggestion
<!-- This checklist can be used as a help for the reviewer. -->
```
Maybe you want this shown in the PR though?

File: .github/pull_request_template.md [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+
+- [ ] Is the code easy to read and understand?
+- [ ] Are comments for humans to read, not computers to disregard?
+- [ ] Does a new feature has an accompanying new test in the CI/CD?
```
Comment (CasperWA) [inline .github/pull_request_template.md:17]:
```suggestion
- [ ] Does a new feature has an accompanying new test (in the CI or unit testing schemes)?
```

File: .github/pull_request_template.md [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+- [ ] Has the documentation been updated as necessary?
+- [ ] Does this close the issue?
+- [ ] Is the change limited to the issue?
+- [ ] Are errors handled for all outcomes?
+- [ ] Does the new feature provide new restrictions on dependecies, and if so is this documented?
```
Comment (CasperWA) [inline .github/pull_request_template.md:22]:
```suggestion
- [ ] Does the new feature provide new restrictions on dependencies, and if so is this documented?
```

**Review by CasperWA — CHANGES_REQUESTED** [review]
By wrapping lines in `<!-- this is an HTML comment --->` they become invisible when creating the PR, even though they have not been removed from the text.

It might be nice to underline that the user making the PR should _not_ interact with the checklist whatsoever, but that it's a reviewer-only feature?

Hmm... actually, the checklist part shouldn't really be in the PR template, but rather it should be a template for the review response (like this one 😉) - also the "Comments" section. But unfortunately, I don't think GitHub supports those kind of templates :|

**Comment by francescalb** [comment]
Yes, initally I wanted a template for the reviewer, but github does not support that as far as I can see. I think this should be an OK help for both reviewer and developer like this, and I think it is better to have this here, than having a separate checklist somewhere else that both reviewer and requester might easily forget.

File: .github/pull_request_template.md (hunk 5 above)
Comment (francescalb) [reply .github/pull_request_template.md:13]: Yes, this should stay visible I think.

**Review by CasperWA — APPROVED** [review]
Looks fine to me.
There might still be a bit of confusion on my end about why the person opening a PR needs to fill out the checklist (as it's written now), since this should be the reviewer's task, but it's minor.

**Comment by CasperWA** [comment]
As a note, you can combine the suggested changes you want to implement from a review and apply/commit them as a batch (all in one) :)

**Comment by francescalb** [comment]
> Looks fine to me.
> There might still be a bit of confusion on my end about why the person opening a PR needs to fill out the checklist (as it's written now), since this should be the reviewer's task, but it's minor.

I kind of agree, this way there is a bit more pressure on the person opening the PR to do it well. Let us try this and see how it works.

**Pull request merged**